context/app/static/js/helpers/functions.ts

```diff
@@ -238,3 +239,43 @@ export function getEntityIcon(entity: { entity_type: ESEntityType; is_component?
   }
   return entityIconMap[entity.entity_type];
 }
+
+/**
+ * Find the creation information for a given entity. Datasets use the published date if available,
+ * otherwise the last modified date. Other entities use the creation date.
+ * @author Austen Money
+ */
+export function getEntityCreationInfo({
+  entity_type,
+  published_timestamp,
+  created_timestamp,
+  last_modified_timestamp,
+}: Pick<Entity, 'entity_type'> &
+  Partial<Pick<Dataset, 'published_timestamp' | 'last_modified_timestamp' | 'created_timestamp'>>) {
+  let creationLabel;
+  let creationVerb;
+  let creationTimestamp;
+
+  if (entity_type === 'Dataset') {
+    if (published_timestamp) {
+      creationLabel = 'Publication Date';
+      creationVerb = 'Published';
+      creationTimestamp = published_timestamp;
+    } else {
+      creationLabel = 'Last Modified';
+      creationVerb = 'Modified';
+      creationTimestamp = last_modified_timestamp;
+    }
```

Comment:
Should these cases also apply for publications?

Reply:
I believe publications use `publication_date` instead of `published_timestamp` - this update doesn't touch the logic on the publication pages. But this function could be updated to handle publications as well, if we wanted to show the `last_modified` or `created` timestamps in the event that the `publication_date` is not provided.